### DISCUSSION TITLE
react-markdown: Option for rendering single paragraph content?

### DISCUSSION THREAD

**Discussion by colepeters** [remarkjs/remark#604] [Q&A] [answered]
I'm just getting up to speed with react-markdown, and wondering if there’s a way to render some arbitrary markdown without any wrapping nodes?

For example, with [markdown-it](https://github.com/markdown-it/markdown-it), there’s [a `renderInline` method](https://markdown-it.github.io/markdown-it/#MarkdownIt.renderInline) for rendering the _contents_ of a single paragraph, but not the wrapping `<p>` node. Could someone give me a hand with figuring out how to do this with react-markdown? I'm not seeing a clear way to do this from the docs.

Thanks in advance!

**Comment by wooorm** [comment, marked as answer]
It might be good to hear more about your use case! Sometimes these things are XY problems, so hearing about what you're actually trying to achieve can help!

**Reply by colepeters** [reply]
Sure! The case I’m working with is we have a React app that renders some text fields (pulled from an Active Admin app) where the content authors use Markdown syntax to mark certain passages of text with bold, italics, and other inline-formatting options. This content is only parsed as Markdown on the front end, and in some instances we want to render the content into specific React components, without anything else wrapping that content.

So on the front end, we might receive the string: `'Here is some text **and part of it is bold**.'`

In our React app, we might have a `Text` component that we want to render this markdown content in. Currently, with markdown-it’s `renderInline`, we'd do this:

```jsx
<Text dangerouslySetInnerHtml={{ __html: renderInline(content)}} />
```

…which results in:

```jsx
<Text>
  Here is some text <strong>and part of it is bold</strong>.
</Text>
```

So what I’m hoping is that react-markdown can take this string of content, apply any inline transformations, and return that content without anything wrapping it as a whole, if that makes sense.

**Reply by wooorm** [reply]
What if someone uses a heading? '## whatever'?

**Reply by colepeters** [reply]
In this instance, block-level syntax is ignored, and not parsed into HTML. [Here’s a basic example](https://codesandbox.io/s/fervent-platform-tmlrj?file=/src/App.js).

**Reply by wooorm** [reply]
Could you post the expected output here? I'm struggling to understand if you want the hashes as text or not. Codesandbox is appreciated but this one either doesn't work at all or it doesn't work on mobile, thanks!

**Reply by colepeters** [reply]
Apologies, for some reason my changes on the sandbox didn't save. I've checked the link again and it should be working now.

Just in case it's not, I'd expect this input:

```
Here is some text. **I am bold.** ## A heading??
```

…with the above setup to result in this:

```jsx
<Text>
  Here is some text. <strong>I am bold.<strong> ## A heading??
</Text>
```

**Reply by wooorm** [reply]
Thanks! I meant, what should happen if there is only `## A heading`? I'm struggling to understand if you want the hashes as text or not.

**Reply by colepeters** [reply]
The hashes can remain as text, yes — any block level formatting would be ignored, and get parsed as plain text.

**Reply by colepeters** [reply]
Hi @wooorm — just wanted to follow up and see if you had any ideas on this? Thanks a lot!

**Reply by wooorm** [reply]
sorry, I don’t really have a good way to solve this. The big thing I see is what would happen if a “paragraph” had a blank line in it? `text\n\nmore text`? That’d be an “impossible” paragraph.

We recently had a discussion about it btw, here: https://github.com/remarkjs/remark/issues/552.

I previously went with using a sanitation schema: https://github.com/remarkjs/remark/issues/552#issuecomment-716367777, which I kinda like. It would drop the hashes though.

**Reply by colepeters** [reply]
Interesting, thanks for the links! The line breaks in this case are fine to skip, as this inline rendering would only be used for single-paragraph content (this is how [markdown-it's `renderInline` function](https://markdown-it.github.io/markdown-it/#MarkdownIt.renderInline) is meant to work, too).

I'm not clear on how your sanitation schema is integrated into the use of react-markdown. Do you happen to have an example you could link me to?

**Reply by wooorm** [reply]
> I'm not clear on how your sanitation schema is integrated into the use of react-markdown.

Unfortunately it doesn’t. We want to enable the rehype ecosystem and its plugins here soon. For now, [react-remark](https://github.com/remarkjs/react-remark) is similar to react-markdown, but *does* allow rehype plugins (and hence such schemas)

**Reply by colepeters** [reply]
Ahh, I see. Thank you! Looks like I might have to put this on hold for now, but great to know this might be coming.
